pkg/api/v1alpha1/tinkerbelldatacenterconfig_webhook.go

```diff
@@ -42,15 +46,49 @@ var _ webhook.Validator = &TinkerbellDatacenterConfig{}
 func (r *TinkerbellDatacenterConfig) ValidateCreate() error {
 	tinkerbelldatacenterconfiglog.Info("validate create", "name", r.Name)
 
-	// TODO(user): fill in your validation logic upon object creation.
+	if err := r.Validate(); err != nil {
+		return apierrors.NewInvalid(
+			GroupVersion.WithKind(TinkerbellDatacenterKind).GroupKind(),
+			r.Name,
+			field.ErrorList{
+				field.Invalid(field.NewPath("spec"), r.Spec, err.Error()),
+			},
+		)
+	}
+
+	if r.IsReconcilePaused() {
```

Comment:
Do we also need to validate that the machine config has a template defined like with vSphere?

Reply:
This webhook is for the datacenter config for Tinkerbell, not the machine config. We can have that check for the machine config webhook.

Reply:
Okay, thanks. I misread the file name!